localstack/services/s3/utils.py

```diff
@@ -85,19 +85,13 @@
     + r"(^(([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])\.)*([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])$)"
 )
 
-REGION_REGEX = r"[a-z]{2}-[a-z]+-[0-9]{1,}"
-PORT_REGEX = r"(:[\d]{0,6})?"
-
 TAG_REGEX = re.compile(r"^[\w\s.:/=+\-@]*$")
 
-S3_VIRTUAL_HOSTNAME_REGEX = (  # path based refs have at least valid bucket expression (separated by .) followed by .s3
-    r"^(http(s)?://)?((?!s3\.)[^\./]+)\."  # the negative lookahead part is for considering buckets
-    r"(((s3(-website)?\.({}\.)?)localhost(\.localstack\.cloud)?)|(localhost\.localstack\.cloud)|"
-    r"(s3((-website)|(-external-1))?[\.-](dualstack\.)?"
-    r"({}\.)?amazonaws\.com(.cn)?)){}(/[\w\-. ]*)*$"
-).format(
-    REGION_REGEX, REGION_REGEX, PORT_REGEX
+
+S3_VIRTUAL_HOSTNAME_REGEX = (
+    r"(?P<bucket>.*).s3.(?P<region>(?:us-gov|us|ap|ca|cn|eu|sa)-[a-z]+-\d)?.*"
 )
```

Comment:
No I think it's the contrary, it actually allows a lot more matches, especially on the domain.
About `s3-website`, that's true, I didn't think of it because for the newer providers, we use a router. I can keep a version of this regex for now for the old provider and just use the new one for `v3` until we remove legacy? I think that could be safer just in case.

Reply:
I don't know to be honest. Is it going to break the legacy provider? If so, then it's too early and we either have to wait with this PR or have an intermediate solution, because we'll remove it with 3.0, but not before that...

Reply:
Agreed! What I've done is use the old `legacy` utils with the old regex for the legacy provider, so that it wouldn't change functionality. We can then remove it easily once it's done.